Field name ‘error’ can not access in reflection code.

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 11.4, 20F71
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)

getwd: /Users/takuya/Works/dev/v-project/mnpp
vmodules: /Users/takuya/.vmodules
vroot: /Users/takuya/Works/dev/v
vexe: /Users/takuya/Works/dev/v/v
vexe mtime: 2021-08-15 08:08:44
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 b72d1e5.eef7eea

Git version: git version 2.30.1 (Apple Git-130)
Git vroot status: weekly.2021.32-56-geef7eea7 (5 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg struct-has-field-named-error.v`
```v

struct Foo {
	error string
	// crror string		// O.K.
}

foo := Foo {
	error: 'this is a error message'
	// crror: 'this is a crror message'	// O.K.
}

$for field in Foo.fields {
	$if field.typ is string {
		println(foo.$(field.name))
	}
}


```

**What did you expect to see?**

Compile successfully


**What did you see instead?**
```
==================
/tmp/v/struct-has-field-named-error.11761043641844909421.tmp.c:10239:17: error: no member named 'error' in 'struct main__Foo'
                                println(foo.error);
                                        ~~~ ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

In the generated C code, the definition of Foo was as follows, so it seems that `error` was treated as a keyword and escaped.

```c
struct main__Foo {
        string _v_error;
};
```

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -20,7 +20,7 @@ fn (mut g Gen) comptime_selector(node ast.ComptimeSelector) {
 		if node.field_expr.expr is ast.Ident {
 			if node.field_expr.expr.name == g.comp_for_field_var
 				&& node.field_expr.field_name == 'name' {
-				g.write(g.comp_for_field_value.name)
+				g.write(c_name(g.comp_for_field_value.name))
 				return
 			}
 		}

--- a/vlib/v/tests/comptime_for_over_struct_with_C_reserved_word_fields_test.v
+++ b/vlib/v/tests/comptime_for_over_struct_with_C_reserved_word_fields_test.v
@@ -1,0 +1,25 @@
+struct StructWithCReservedWord {
+	error  string
+	while  int
+	extern int
+	switch bool
+}
+
+fn test_structs_that_have_fields_that_are_reserved_c_words_can_be_iterated() {
+	foo := StructWithCReservedWord{
+		error: 'this is an error message'
+		while: 123
+		extern: 456
+		switch: true
+	}
+	$for field in StructWithCReservedWord.fields {
+		$if field.typ is string {
+			println(foo.$(field.name))
+		}
+	}
+	assert foo.switch
+	assert foo.extern == 456
+	assert foo.while == 123
+	assert foo.error == 'this is an error message'
+	println(foo)
+}

